node/blockAPI/metaBlock_test.go

```diff
@@ -0,0 +1,282 @@
+		storerMock,
+		true,
+		false,
+	)
+
+	header := &block.MetaBlock{
+		Nonce: nonce,
+		Round: round,
+		Epoch: epoch,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: miniblockHeader,
+				Type: block.TxBlock,
+			},
+		},
+		AccumulatedFees:        big.NewInt(0),
+		DeveloperFees:          big.NewInt(0),
+		AccumulatedFeesInEpoch: big.NewInt(10),
+		DevFeesInEpoch:         big.NewInt(5),
+	}
+	headerBytes, _ := json.Marshal(header)
+	_ = storerMock.Put(headerHash, headerBytes)
+
+	nonceBytes := uint64Converter.ToByteSlice(nonce)
+	_ = storerMock.Put(nonceBytes, headerHash)
+
+	expectedBlock := &api.Block{
+		Nonce:           nonce,
+		Round:           round,
+		Shard:           core.MetachainShardId,
+		Epoch:           epoch,
+		Hash:            hex.EncodeToString(headerHash),
+		NotarizedBlocks: []*api.NotarizedBlock{},
+		MiniBlocks: []*api.MiniBlock{
+			{
+				Hash: hex.EncodeToString(miniblockHeader),
+				Type: block.TxBlock.String(),
+			},
+		},
+		AccumulatedFees:        "0",
+		DeveloperFees:          "0",
+		AccumulatedFeesInEpoch: "10",
+		DeveloperFeesInEpoch:   "5",
+		Status:                 BlockStatusOnChain,
+	}
+
+	blk, err := metaAPIBlockProcessor.GetBlockByHash(headerHash, false)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedBlock, blk)
+}
+
+func TestMetaAPIBlockProcessor_GetBlockByNonceFromHistoryNode(t *testing.T) {
+	t.Parallel()
+
+	nonce := uint64(1)
+	round := uint64(2)
+	epoch := uint32(1)
+	miniblockHeader := []byte("miniBlockHash")
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+
+	metaAPIBlockProcessor := createMockMetaAPIProcessor(
+		headerHash,
+		storerMock,
+		true,
+		false,
+	)
+
+	header := &block.MetaBlock{
+		Nonce: nonce,
+		Round: round,
+		Epoch: epoch,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: miniblockHeader,
+				Type: block.TxBlock,
+			},
+		},
+		AccumulatedFees:        big.NewInt(0),
+		DeveloperFees:          big.NewInt(0),
+		AccumulatedFeesInEpoch: big.NewInt(10),
+		DevFeesInEpoch:         big.NewInt(5),
+	}
+
+	headerBytes, _ := json.Marshal(header)
+	_ = storerMock.Put(headerHash, headerBytes)
+
+	expectedBlock := &api.Block{
+		Nonce:           nonce,
+		Round:           round,
+		Shard:           core.MetachainShardId,
+		Epoch:           epoch,
+		Hash:            hex.EncodeToString(headerHash),
+		NotarizedBlocks: []*api.NotarizedBlock{},
+		MiniBlocks: []*api.MiniBlock{
+			{
+				Hash: hex.EncodeToString(miniblockHeader),
+				Type: block.TxBlock.String(),
+			},
+		},
+		AccumulatedFees:        "0",
+		DeveloperFees:          "0",
+		AccumulatedFeesInEpoch: "10",
+		DeveloperFeesInEpoch:   "5",
+		Status:                 BlockStatusOnChain,
+	}
+
+	blk, err := metaAPIBlockProcessor.GetBlockByNonce(1, true)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedBlock, blk)
+}
+
+func TestMetaAPIBlockProcessor_GetBlockByHashFromHistoryNodeStatusReverted(t *testing.T) {
```

Comment:
maybe you can add a test for a block that contains transactions

Reply:
will be added after unmarshallTx refactor. covered in baseBlock tests for now